src/Language/Haskell/Liquid/Parse.hs

```diff
@@ -750,13 +749,18 @@ tyBodyP ty
           outTy (RFun _ _ t _) = Just t
           outTy _              = Nothing
 
+upperIdP' :: Parser Symbol
+upperIdP' = try $ symbol <$> condIdP alphanums (isUpper . head)
+  where
+    alphanums = ['A' .. 'Z'] ++ ['a'..'z'] ++ ['0'..'9'] ++ "'"
```

Comment:
Won't this allow parsing things like `'Foo`, which is not a valid Haskell name?

Also, we should also accept `'`s in identifiers, not just constructors, so it would probably be best to have one parser that accepts *any* valid Haskell name, and a variation that checks that the first character is upper-case.